client-misk-jooq/src/test/kotlin/app/cash/backfila/client/misk/MiskJooqBackfillTests.kt

```diff
@@ -0,0 +1,497 @@
+      backfillClass = JooqWidgetCompoundKeyBackfill::class,
+      backfillRowKeys = WidgetCompoundKeyBackfillDbSetup.createSome(transacter, clock, tokenGenerator),
+      backfila
+    )
+  ).map { backfillOption ->
+    DynamicTest.dynamicTest("wetAndDryRunProcessSameElements test ${backfillOption.backfillClass.simpleName}") {
+      val dryRun = backfillOption.createDryRun()
+        .apply { configureForTest() }
+      dryRun.execute()
+      app.cash.backfila.client.misk.testing.assertThat(dryRun).isComplete()
+      assertThat(dryRun.backfill.idsRanWet).isEmpty()
+
+      val wetRun = backfillOption.createWetRun()
+        .apply { configureForTest() }
+      wetRun.execute()
+      app.cash.backfila.client.misk.testing.assertThat(wetRun).isComplete()
+      assertThat(wetRun.backfill.idsRanDry).isEmpty()
+
+      assertThat(dryRun.backfill.idsRanDry).containsExactlyElementsOf(wetRun.backfill.idsRanWet)
+    }
+  }
+
+  @TestFactory
+  fun checkForDescriptions() = listOf(
+    JooqBackfillTestOptions(
+      backfillClass = JooqMenuTestBackfill::class,
+      backfillRowKeys = JooqMenuBackfillDbSetup.createSome(transacter),
+      backfila,
+      description = "So we can backfill menus."
+    ),
+    JooqBackfillTestOptions(
+      backfillClass = JooqWidgetCompoundKeyBackfill::class,
+      backfillRowKeys = WidgetCompoundKeyBackfillDbSetup.createSome(transacter, clock, tokenGenerator),
+      backfila,
+      description = "So we can backfill widgets."
+    )
+  ).map { backfillOption ->
+    DynamicTest.dynamicTest("checkForDescriptions test ${backfillOption.backfillClass.simpleName}") {
+      val dryRun = backfillOption.createDryRun()
+        .apply { configureForTest() }
+      dryRun.execute()
+      app.cash.backfila.client.misk.testing.assertThat(dryRun).isComplete()
+      assertThat(dryRun.backfill.idsRanWet).isEmpty()
+
+      val wetRun = backfillOption.createWetRun()
+        .apply { configureForTest() }
+      wetRun.execute()
+      app.cash.backfila.client.misk.testing.assertThat(wetRun).isComplete()
+      assertThat(wetRun.backfill.idsRanDry).isEmpty()
+
+      assertThat(dryRun.backfill.idsRanDry).containsExactlyElementsOf(wetRun.backfill.idsRanWet)
+    }
+  }
+
+  @TestFactory
+  fun checkParameters() = listOf(
+    JooqBackfillTestOptions(
+      backfillClass = JooqMenuTestBackfill::class,
+      backfillRowKeys = JooqMenuBackfillDbSetup.createSome(transacter),
+      backfila,
+      parameterData = mapOf("type" to "beef".encodeUtf8())
+    ),
+    JooqBackfillTestOptions(
+      backfillClass = JooqWidgetCompoundKeyBackfill::class,
+      backfillRowKeys = WidgetCompoundKeyBackfillDbSetup.createSome(transacter, clock, tokenGenerator),
+      backfila,
+      parameterData = mapOf("manufacturerTokensCSV" to "token2 - not select for backfill".encodeUtf8())
+    )
+  ).map { backfillOption ->
+    DynamicTest.dynamicTest("checkParameters test ${backfillOption.backfillClass.simpleName}") {
```

Comment:
I assume you copied these test cases from my hibernate version for the most part?

Reply:
yep. My idea was to test that the jooq backfill was an exact drop in replacement for the hibernate one.